From 7ed6cf97399420e118ad1ad68af94f6d0a604902 Mon Sep 17 00:00:00 2001
From: Carsten Haitzler (Raster <raster@rasterman.com>
Date: Thu, 30 Oct 2008 16:47:23 +1100
Subject: [PATCH] palmt650.conf: add palmt650-keyboard-modmap package to make keyboard work in x

the x keymap for the palmt650 is mostly wrong missing almost all the useful
keys. this provides a package that sets a modifier map when starting x to
set all the keys up properly for the palmt650.
---
 packages/palmt650/palmt650-keyboard-modmap.bb      |   23 ++++
 .../palmt650/palmt650-keyboard-modmap/60xXmodmap   |    8 ++
 packages/palmt650/palmt650-keyboard-modmap/Xmodmap |  118 ++++++++++++++++++++
 3 files changed, 149 insertions(+), 0 deletions(-)
 create mode 100644 packages/palmt650/palmt650-keyboard-modmap.bb
 create mode 100755 packages/palmt650/palmt650-keyboard-modmap/60xXmodmap
 create mode 100644 packages/palmt650/palmt650-keyboard-modmap/Xmodmap

diff --git a/packages/palmt650/palmt650-keyboard-modmap.bb b/packages/palmt650/palmt650-keyboard-modmap.bb
new file mode 100644
index 0000000..dde7fb4
--- /dev/null
+++ b/packages/palmt650/palmt650-keyboard-modmap.bb
@@ -0,0 +1,23 @@
+DESCRIPTION = "A X keyboard modifier mapping for the Treo650"
+LICENSE = "GPL"
+PV = "0.0.3"
+PR = "r0.01"
+
+SRC_URI = "\
+  file://60xXmodmap \
+  file://Xmodmap \
+"
+
+S = "${WORKDIR}"
+
+do_install() {
+  install -d ${D}/etc/X11/
+  install ${WORKDIR}/Xmodmap ${D}/etc/X11/Xmodmap
+  install -d ${D}/etc/X11/Xsession.d
+  install ${WORKDIR}/60xXmodmap ${D}/etc/X11/Xsession.d/60xXmodmap
+}
+
+PACKAGE_ARCH = "all"
+
+FILES_${PN} += "${bindir}/* ${libdir}/lib*.so.*"
+FILES_${PN} += "/etc/X11/Xsession.d/80x-enlightenment-wakerd"
diff --git a/packages/palmt650/palmt650-keyboard-modmap/60xXmodmap b/packages/palmt650/palmt650-keyboard-modmap/60xXmodmap
new file mode 100755
index 0000000..db814a0
--- /dev/null
+++ b/packages/palmt650/palmt650-keyboard-modmap/60xXmodmap
@@ -0,0 +1,8 @@
+# -*- sh -*-
+# Xsession.d script for Xmodmap
+#
+#
+# This file is sourced by Xsession(5), not executed.
+
+xmodmap /etc/X11/Xmodmap
+xmodmap -e 'remove Control = Mode_switch'
diff --git a/packages/palmt650/palmt650-keyboard-modmap/Xmodmap b/packages/palmt650/palmt650-keyboard-modmap/Xmodmap
new file mode 100644
index 0000000..9086cd1
--- /dev/null
+++ b/packages/palmt650/palmt650-keyboard-modmap/Xmodmap
@@ -0,0 +1,118 @@
+!!!!! get modifiers right
+remove Control = Mode_switch
+remove Control = Shift_R
+remove Shift = Menu
+remove Shift = Control_R
+! Alt key -> Alt_R
+keycode 108 = Alt_R
+! Shift(R) key -> Shift_R
+!keycode 105 = Shift_R
+!add Shift = Shift_R
+! hack - should be Control_R for terminal
+keycode 105 = Control_R
+add Control = Control_R
+! blue key -> Mode_switch
+keycode 37 = Mode_switch
+add mod5 = Mode_switch
+
+!!!!! top mute/unmute
+! keycode mute = has no keycode
+
+!!!!! side buttons
+keycode 123 = Prior
+keycode 122 = Next
+keycode 71 = End
+
+!!!!! top 2 buttons left/right and up from arrow key nav
+!! below - added hacked symbols: percent asciicircum
+keycode 69 = Home percent asciicircum
+!! below - added hacked symbols: percent asciicircum
+keycode 9 = Menu Delete Insert
+
+!!!!! arrowkeys
+keycode 111 = Up
+keycode 116 = Down
+keycode 113 = Left
+keycode 114 = Right
+keycode 104 = KP_Enter
+
+!!!!! answer, calendar, mail, hangup/power
+!!!! CHECKME
+!! below - added hacked symbols: Escape Print
+keycode 72 = F1 Escape Print
+!! below - added hacked symbols: equal semicolon
+keycode 67 = F2 equal semicolon
+!! below - added hacked symbols: bracketleft braceleft
+keycode 68 = F3 bracketleft braceleft
+!! below - added hacked symbols: bracketright braceright
+keycode 70 = F4 bracketright braceright
+
+!!!!! 1st row
+keycode 24 = q Q slash
+keycode 25 = w W plus
+keycode 26 = e E 1
+keycode 27 = r R 2
+keycode 28 = t T 3
+keycode 29 = y Y parenleft
+keycode 30 = u U parenright
+keycode 31 = i I at
+keycode 32 = o O quotedbl
+!! below - added hacked symbols: underscore
+keycode 33 = p P underscore
+!!!!! 2nd row
+keycode 38 = a A ampersand
+keycode 39 = s S minus
+keycode 40 = d D 4
+keycode 41 = f F 5
+keycode 42 = g G 6
+keycode 43 = h H dollar
+keycode 44 = j J exclam
+keycode 45 = k K colon
+keycode 46 = l L apostrophe
+!! below - added hacked symbols: backslash asciitilde
+keycode 22 = BackSpace backslash asciitilde
+!!!!! 3rd row
+keycode 37 = Mode_switch
+keycode 52 = z S asterisk
+keycode 53 = x D 7
+keycode 54 = c F 8
+keycode 55 = v G 9
+keycode 56 = b H numbersign
+keycode 57 = n J question
+keycode 58 = m K comma
+!! below - added hacked symbols: bar grave
+keycode 60 = period bar grave
+keycode 36 = Return
+!!!!! bottom row
+keycode 50 = Shift_L
+!! below - added hacked symbols: less greater
+keycode 19 = 0 less greater
+keycode 65 = space
+keycode 108 = Alt_R
+!! below - hacked to be Control_R instead of Shift_R
+keycode 105 = Control_R
+
+!!!!! missing standard key symbols:
+! % hacked
+! ^ hacked
+! _ hacked
+! = hacked
+! [ hacked
+! ] hacked
+! { hacked
+! } hacked
+! ; hacked
+! < hacked
+! > hacked
+! \ hacked
+! | hacked
+! ` hacked
+! ~ hacked
+! Escape hacked
+! Delete hacked
+! Insert hacked
+! Print hacked
+! Pause
+! Caps_Lock
+! Num_Lock
+! Sroll_Lock
-- 
1.5.4.3

